### PR TITLE
(shortfin-sd) Program initialization and logging improvements

### PR DESCRIPTION
**Pull request by monorimet**

### CODE REVIEW

**Comment by monorimet** [comment]
Best found throughput number (11/6) is at .89 samples per second per gpu. More progress to come.

File: shortfin/dev_me.py [hunk 1]
```diff
@@ -196,7 +197,8 @@ def configure_mode(env_info: EnvInfo, args):
         env_vars["SHORTFIN_IREE_SOURCE_DIR"] = env_info.iree_dir
     if env_info.clang_exe:
         env_vars["CC"] = env_info.clang_exe
-        env_vars["CXX"] = f"{env_info.clang_exe}++"
+        clangxx_exe = "clang++-".join(env_info.clang_exe.split("clang-"))
```
Comment (stellaraccident) [inline shortfin/dev_me.py:200]: Does this work for unadorned "clang" commands? Maybe drop this file from the patch and we do it separately?

File: shortfin/python/shortfin_apps/sd/components/manager.py [hunk 2]
```diff
@@ -34,7 +34,7 @@ def __init__(self, device="local-task", device_ids=None):
                         raise ValueError(f"Device id {did} could not be parsed.")
             else:
                 selected = available
-            sb = sf.amdgpu.SystemBuilder(amdgpu_visible_devices=";".join(selected))
+            sb = sf.SystemBuilder(system_type="amdgpu")
```
Comment (stellaraccident) [inline shortfin/python/shortfin_apps/sd/components/manager.py:37]: You can just create the SystemBuilder once:

```
sb = sf.SystemBuilder(system_type="amdgpu")
sb.visible_devices = get_selected_devices(sb)
self.ls = sb.create_system()
```

(i.e. you can set properties between construction of the SystemBuilder and create_system())

Note that the property takes a normal list of strings so no need to join.

It also looks like you dropped the visible_devices thing in this patch.

File: shortfin/src/shortfin/local/systems/amdgpu.cc [hunk 3]
```diff
@@ -23,6 +23,7 @@ AMDGPUSystemBuilder::AMDGPUSystemBuilder(iree_allocator_t host_allocator,
   iree_hal_hip_device_params_initialize(&default_device_params_);
   InitializeDefaultSettings();
   config_options().ValidateUndef();
+  default_device_params_.async_allocations = false;
```
Comment (stellaraccident) [inline shortfin/src/shortfin/local/systems/amdgpu.cc:26]: Can drop this now.

**Review by stellaraccident — APPROVED** [review]
Some nits.

File: shortfin/python/shortfin_apps/sd/components/manager.py (hunk 2 above)
Comment (monorimet) [reply shortfin/python/shortfin_apps/sd/components/manager.py:37]: yeah, that was for making sure the specified device IDs actually matched an available device. I don't really like it, so I'll pass in the raw args.

**Pull request merged**